numpy/random/mtrand/generate_randint_helpers.py

```diff
@@ -0,0 +1,152 @@
+"""
+Helper file for generating the necessary helper functions
+for each `dtype` in `np.random.randint`. The function
+`gen_randint_helpers` is called right before Cythonization
+in NumPy setup.
+
+"""
+
+# npy_dt : Desired C API data type
+# npy_udt : Unsigned version of desired C
+#           API data type (could be identical)
+# npy_dt : Desired Python API data type
+
+template = """
+def _rand_{npy_dt}(low, high, size, rngstate):
```

Comment:
I should add once you cdef you can directly pass the pointer to the state using its C-type, and so the modification needed to encapsulate are not necessary. 

See, for example, 

https://github.com/bashtage/ng-numpy-randomstate/blob/master/randomstate/bounded_integers.pxi#L12